sentence_transformers/util.py

```diff
@@ -655,18 +658,27 @@ def mine_hard_negatives(
 
     log_counters = {}
     queries = dataset[columns[0]]
-    corpus = dataset[columns[1]]
+    positives = dataset[columns[1]]
+    separate_corpus = corpus is not None
+    if not separate_corpus:
+        corpus = positives
```

Comment:
I've simplified this to just a list of strings. For `dataset` it makes sense for it to be a 2-column Dataset (in my opinion), as that is a common format that is often used to train embedding models, which can be easily loaded. See for example this collection: https://huggingface.co/collections/sentence-transformers/embedding-model-datasets-6644d7a3673a511914aa7552

A lot of these datasets have 2 columns.

But if you also have a collection of negative candidates, then it might be a hastle to turn that into a Dataset?

Reply:
I totally agree, thanks for the clarification!